### PR TITLE
#4418 Installer support for deployment options

### LINKED ISSUE #4418: DB Support Slice 4: In Extension, add Deployment Installer Support for Blueprint Options
\[Note: This issue was auto\-generated from Product Board\]

<p></p>

See spec and comments in spec:

<https://www.notion.so/pixiebrix/Activation-time-Database-Selection-800b2548093d415d981ce40d656489c7>

<p></p>

### 4\. In Extension, add Deployment Installer Support for Blueprint Options

- In deployments\.ts, modify the call to installRecipe to pass the optionsArgs from options\_config

- In useDeployments, modify the call to installRecipe to pass the optionsArgs from options\_config

<p></p>

### PR DESCRIPTION
**Pull request by BALEHOK**
## What does this PR do?

- Closes #4418 
- ~~Depends on https://github.com/pixiebrix/pixiebrix-app/issues/2394 and will need to get the latest `swagger.ts` update~~ - Done

## Checklist

- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/types/swagger.ts [hunk 1]
```diff
@@ -727,6 +729,7 @@ export interface components {
         auth: string;
       }[];
       active?: boolean;
+      options_config?: Record<string, Primitive>;
```
Comment (BALEHOK) [inline src/types/swagger.ts:732]: This matches the type used in [action](https://github.com/pixiebrix/pixiebrix-extension/blob/0a0531b418b2e6231c51a7da5b4fea1c78d4b874/src/store/extensionsSlice.ts#L104)

Comment (BLoe) [reply src/types/swagger.ts:732]: It looks like we use the type `{ [key: string]: unknown }` elsewhere in this swagger file. Are you sure you're allowed to use `Record`, `Primitive`, etc. types in the swagger config?

Comment (mnholtz) [reply src/types/swagger.ts:732]: This change should be generated by our changes on the back-end. We'll implement the next slice and update accordingly.

Comment (BLoe) [reply src/types/swagger.ts:732]: @BALEHOK Sounds like we should probably just leave this open and wait for the back end slice to be merged, then rebase this one on top of that so we don't have to make the swagger change in this PR.

File: src/types/swagger.ts [hunk 2]
```diff
@@ -1424,11 +1417,19 @@ export interface components {
...
+      id?: number;
+      /** @description Just the error message, not the complete traceback */
       message: string;
+      /** @description Step of the extension, depends on the extension's telemetry settings */
+      step_label?: string | null;
+      /**
+       * Format: date-time
+       * @description Timestamp the error occurred, not the time the record is added to the db
+       */
+      timestamp: string;
```
Comment (mnholtz) [inline src/types/swagger.ts:1432]: All these extra `swagger.ts` changes look good to me. No worries there. This file just sometimes drifts out of sync with the app when we make admin-console specific api changes.

**Comment by mnholtz** [comment]
This looks good to me. Do we need to manually test/add tests at this time?

I'm having trouble figuring out why I lost ability to "approve" this PR. @BALEHOK would you mind re-requesting review?

File: src/types/swagger.ts (hunk 1 above)
Comment (mnholtz) [reply src/types/swagger.ts]: @BALEHOK swagger file up-to-date. This is good to go for a full review.

**Comment by BALEHOK** [comment]
@BLoe @mnholtz  I pulled the latest `swagger.ts` from the App, should be good for the final review now.
You might want to see the changes to [swagger](https://github.com/pixiebrix/pixiebrix-app/pull/2414/files#diff-52a3555f2455eea35c1cb91b52bf417be00c80ba1c30812bca83312446ac5d29) and [contracts](https://github.com/pixiebrix/pixiebrix-app/pull/2414/files#diff-ef0d9e7ba4d71c3bb2283ad030d8b0a4022295ec71a2cc07a6984eaa577257c0) files in the App.

File: src/core.ts [hunk 3]
```diff
@@ -930,7 +929,7 @@ export interface IconConfig {
   color?: string;
 }
 
-export type UserOptions = Record<string, Primitive>;
+export type UserOptions = Record<string, unknown>;
```
Comment (BLoe) [inline src/core.ts:932]: Just curious, what's the need for this change?

**Review by BLoe — APPROVED** [review]

File: src/core.ts (hunk 3 above)
Comment (BALEHOK) [reply src/core.ts:932]: It's just that `Record<string, unknown>;` better aligns with the API response for the `Deployment.optionsArgs`.

**Pull request merged**